### ISSUE TITLE
~PyTorch Docathon H1 2023~

### ISSUE THREAD

**Issue by svekars** [pytorch/tutorials#2350]
# 🎉 It's a wrap! 🎉

See our [leaderboard](https://github.com/pytorch/tutorials/blob/main/docathon-leaderboard.md) and [blog post](https://pytorch.org/blog/docathon-h1-2023-wrap-up/). Thank you to  everyone who contributed and congrats to the winners! 

We have a large backlog of issues that we want to address and it's a great opportunity for you to start contributing to PyTorch. We have limited this docathon to the [pytorch/tutorials](https://github.com/pytorch/tutorials) and [pytorch/examples](https://github.com/pytorch/examples) repositories, so please work on the issues from these two repositories.

# Date and location 
**WHEN:** The docathon starts on May 31st 10 AM PST. Please do not work on tasks until then. We will continue accepting new submissions until 5 PM PST on June 11th.
**WHERE:** Virtual
**WHAT:** Issues with the **docathon-h1-2023** label - will be posted on May 31.

Watch our intro video to learn more details about the event.

[![Watch the docathon intro](https://github-production-user-asset-6210df.s3.amazonaws.com/5317992/242342554-2a0d5489-0f16-4db0-b3c7-67a9ada9abe6.png)](https://youtu.be/qNAZtYowAM0)

# Can everyone participate?

We encourage everyone to consider participating in the docathon but there are a few things we expect from the participants:

- You must have a GitHub account and know how to use Git and GitHub, how to submit or rebase your PR on the latest main branch, how to fork or clone the repo, how to view errors in the CI and troubleshoot. We reserve the right to reject incorrectly submitted PRs.
- You must be familiar with Python, the basics of Machine Learning, and have at least a basic knowledge of PyTorch. Familiarity with Sphinx, sphinx-gallery, and reStructuredText is a plus.

Before you start contributing make sure to read [Linux Foundation Code of Conduct](https://events.linuxfoundation.org/about/code-of-conduct/).

# What contributions are we looking for?

All issues for this docathon are tagged with the **docathon-h1-2023** label. Please note that contributions that address other issues won't be counted. We are primarily looking for the following contributions: 

**NOTE:** Please avoid working on issues with **intel**, **amd**, and **nvidia** labels which are reserved for our partners.

- Bug fixes in the [pytorch/tutorials](https://github.com/pytorch/tutorials) repo tagged with the docathon-h1-2023 label - see [the list](https://github.com/pytorch/tutorials/issues?q=is%3Aopen+is%3Aissue+label%3Adocathon-h1-2023).
- New examples in the [pytorch/examples](https://github.com/pytorch/examples) repo tagged with the docathon-h1-2023 label - see [the issue](https://github.com/pytorch/examples/issues?q=is%3Aopen+is%3Aissue+label%3Adocathon-h1-2023). 

**NOTE:** Due to the large number of RSVPs, the tasks are provided on a first come first serve basis — please don't hoard the tasks!

# Difficulty Levels

The issues have three levels of difficulty: **easy**, **medium**, and **advanced**. If this is your first time contributing to PyTorch, we recommend that you start with an issue that is tagged as **easy**.

# How to contribute to tutorials?

1. Read [pytorch/tutorials/CONTRIBUTING.md](https://github.com/pytorch/tutorials/blob/main/CONTRIBUTING.md) for general guidelines on how the submission process works and overall style and voice. 
2. Pick an issue that is labeled as **docathon-h1-2023**. 
3. In the issue, add a comment with the text /assigntome. If the issue is already assigned, please find another issue to work on. We ask that you assign one issue at a time - we want to give everyone a fair chance to participate. When you are done with one issue and get it approved, you can assign another one to yourself and start working on it.
4. If you are submitting a new tutorial, use [this template](https://github.com/pytorch/tutorials/blob/main/beginner_source/template_tutorial.py).
5. Fork or clone the PyTorch repository to your computer. For simple fixes, like incorrect URLs, you could use the GitHub UI as well.
6. Create a branch and work on the fix.
7. Test your fix by running the single tutorial locally. Don't run the whole build as it takes hours and requires a GPU. You can run one tutorial as a script python3 <tutorial-name.py> or GALLERY_PATTERN="neural_style_transfer_tutorial.py" make html
8. After you fix all the issues, you are ready to submit your PR.

# Submit Your PR

1. Submit your PR referencing the issue you've picked. For example:

<img width="1058" alt="s_pytorch_pr_example" src="https://github.com/pytorch/tutorials/assets/5317992/f838571a-83d0-4908-94b6-3f7e3b200825">
 
3. If you have not yet, sign the Contributor License Agreement (CLA) - prompted as a check in the PR. We can't accept any PRs without a signed CLA.
4. Watch for any CI errors and fix as needed - all checks must pass successfully. 
5. There are two ways to check the resulting HTML. For simple fixes and .rst files, you can check the HTML output in Netlify that is posted as Netlify bot comment in the PR. For new tutorials and significant rewrites, you need to make sure that the output is generated correct. You can check the generated output for python tutorials in the ARTIFACTS tab in CricleCI:
     1.  In the list of checks, click **Details** on the **ci/circleci: pytorch_tutorial_pr_build_manager** check
          
         <img width="841" alt="s_PR_manager" src="https://github.com/pytorch/tutorials/assets/5317992/fb51d2c5-5867-43a0-bf89-0e6b2d5911d6">

     3. In the CircleCI UI, select the ARTIFACTS tab, open `tutorials/index.html`, and locate you tutorial in this preview:
     
        <img width="830" alt="s_PR_artifacts" src="https://github.com/pytorch/tutorials/assets/5317992/5e7513bd-e373-458f-ac6c-2b717e09b40d">

7. The reviewers might provide feedback that we expect you to address.
8. When all feedback is addressed and your PR is approved - one of the reviewers will merge your PR. 

# How to contribute to examples?

In the examples repo, we are looking to create a few new examples that demonstrate how to create models from scratch. Pre-trained models will not be accepted. Note, that all contributions to the examples repo are advanced and if you are not as familiar with PyTorch, you might want to pick an issue in the pytorch/tutorials repo instead.
There is only one issue in the pytorch/examples repo:

1. Read the [examples/CONTRIBUTING.md](https://github.com/pytorch/examples/blob/main/CONTRIBUTING.md) for general guidelines on the submission.
5. Review the list of new examples we want to create in [this issue](https://github.com/pytorch/examples/issues/1131) and pick a model you want to write an example about.
6. Before starting to write your code, check the comments in [the issue](https://github.com/pytorch/examples/issues/1131) to verify that no one is actively working on the example you picked. If in doubt, ping @msaroufim. 
7. Your submission must be a new folder named after the example, in which you'll have: your Python file implementing a PyTorch model from scratch with no dependencies, requirements.txt, a 1 line test, a README.md and optionally a short update to the pytorch.org/examples page. See [this example](https://github.com/pytorch/examples/pull/1003) for reference and [the issue](https://github.com/pytorch/examples/issues/1131) for more detailed instructions.
8. Tag your PR with the docathon-h1-2023 label to ensure your contribution is counted as a docathon submission. In the description of the PR, link to the issue #1131 like this:  Re: #1131. 
9. Address feedback in your PR and make sure the test passes. 
10. After your PR is merged, delete your branch and mark the model you've described in the checklist in [the issue](https://github.com/pytorch/examples/issues/1131). 

# Can I partner with someone to work on an issue?

Unless you are working on a completely new tutorial from scratch, most of the issues should be possible to address on your own. If you decide to partner with someone, you can find someone to work with on our Slack channel by posting a free-form request to collaborate. One individual from the group can submit a PR referring others as co-authors by specifying their GitHub usernames in the commit message like this:

```
Co-authored-by: NAME <NAME@EXAMPLE.COM>
Co-authored-by: ANOTHER-NAME <ANOTHER-NAME@EXAMPLE.COM>
```

Depending on the complexity of the issue, we reserve the right to decline contributions from multiple co-authors for trivial issues like fixing formatting, broken links, or very small code changes. For all issues that are not new tutorials or examples, please, limit the number of co-authors to two. 

# Top contributors recognition

For all contributions addressing the issues with the **docathon-h1-2023** label merged to the main branch in [pytorch/tutorials](https://github.com/pytorch/tutorials) or [pytorch/examples](https://github.com/pytorch/examples) repos during the period from May 31 to June 15, you will get a special PyTorch GitHub badge. The issues will be released on the first day of the docathon. The top contributors will receive additional recognition and will be featured in a PyTorch social media announcement.

# Questions?
You can post your questions in the docathon Slack channel.

**Comment by kiszk** [comment]
I cannot find the invitation to the slack channel. Could you please send an invitation again? Thanks in advance.

**Comment by svekars** [comment]
@kiszk I just sent an invite. Can you please take a look?

**Comment by kiszk** [comment]
Thanks, got it.

**Comment by justinjk007** [comment]
I would like to be invited to the slack channel as well please ?

**Comment by sunzhongkai588** [comment]
the invitation is invalid. Could you please send an invitation again? Thx

**Comment by Viditagarwal7479** [comment]
Hey, when I am trying to register for the Docathon, LF page is showing that the event is full with a participant count of 242. It's hardly 12hrs since launch. Could anything be done regarding this?
Thanks.
![Screenshot_20230601-100406](https://github.com/pytorch/tutorials/assets/72168180/7c9c430b-f889-4356-8e9b-4128e8939f09)

**Comment by iswarup** [comment]
> @kiszk I just sent an invite. Can you please take a look?

Hi @svekars 
Where do we get these invitations?
I searched my Gmail inbox but couldn't find it.
Please help

**Comment by bjhargrave** [comment]
You may want to fix the main comment where it says 

    In the issue, add a comment with the text \assigntome

to 

    In the issue, add a comment with the text /assigntome

I have seen several issues where the contributor used the backslash variation which was not understood by the bot.

**Comment by carljparker** [comment]
> > @kiszk I just sent an invite. Can you please take a look?
> 
> Hi @svekars Where do we get these invitations? I searched my Gmail inbox but couldn't find it. Please help

@iswarup I'm afraid that you are not on our list of people who registered for this docathon, so you did not receive the invite . . . :-( . . . But we plan to have other docathons in the future. Appreciate your interest.

**Comment by Ilinapradhan** [comment]
I cannot find the invitation to the slack channel. Could you please send an invitation again? Thanks in advance.

**Comment by Ilinapradhan** [comment]
My email id -ilina04khushi@gmail.com

**Comment by svekars** [comment]
you are already in the channel @Ilinapradhan

**Comment by NeoKish** [comment]
@svekars Yeah even I couldn't find the slack channel in the emails from LFHQ

**Comment by svekars** [comment]
@NeoKish just sent you an invite
